packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -132,48 +138,61 @@ export class DataProxyEngine extends Engine {
   }
 
   private async requestInternal<T>(body: Record<string, any>, headers: Record<string, string>, attempt: number) {
-    await this.initPromise
+    await this.pushPromise
 
     try {
       this.logEmitter.emit('info', {
-        message: `Calling ${await this.url('graphql')} (n=${attempt})`,
+        message: `Calling ${this.url('graphql')} (n=${attempt})`,
       })
 
-      const res = await request(await this.url('graphql'), {
+      const res = await request(this.url('graphql'), {
         method: 'POST',
         headers: { ...headers, ...this.headers },
         body: JSON.stringify(body),
       })
 
-      // 404 on the GraphQL, so we re-upload schema & retry
-      if (res.status === 404) {
-        await this.uploadSchema()
+      const err = await responseToError(res)
```

Comment:
This originates from a fix for better error codes https://github.com/prisma/prisma/pull/10085